examples/all-clusters-app/all-clusters-common/BUILD.gn

```diff
@@ -53,7 +53,7 @@ source_set("all-clusters-common") {
     "${chip_root}/src/app/clusters/tv-channel-server/tv-channel-server.cpp",
     "${chip_root}/src/app/reporting/reporting-default-configuration.cpp",
     "${chip_root}/src/app/reporting/reporting.cpp",
-    "${chip_root}/src/app/server/DataModelHandler.cpp",
+    "${chip_root}/src/app/util/DataModelHandler.cpp",
```

Comment:
why the move from server to util?

generally I am reluctant to place things under util/helper/support named things as it becomes a catch-all for everything.
Are we removing 'server' completely? what would be the intended final layout of files?

Reply:
Having it in `src/app/server` doesn't make much sense for several reasons:
1) all other files in `src/app/server` are built as a library defined in `src/app/server/BUILD.gn` while `DataModelHandler.cpp` is included directly by applications (like other Data Model files from `src/app/util`)
2) other files in `src/app/server` are only relevant for the accessory/server side while `DataModelHandler.cpp` is pulled by both controllers and accessories.
If found it quite ugly that we pulled files from `src/app/server` into controller applications to handle client side of the ZCL clusters.